aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -173,179 +175,183 @@ async def create_pres(
+                    # Avoiding addition of duplicate records
+                    (
+                        vcrecord_list,
+                        vcrecord_ids_set,
+                    ) = await self.process_vcrecords_return_list(records, record_ids)
+                    record_ids = vcrecord_ids_set
+                    credentials_list = credentials_list + vcrecord_list
+            except StorageNotFoundError as err:
+                raise V20PresFormatHandlerError(err)
+            except TypeError as err:
+                LOGGER.error(str(err))
+                responder = self._profile.inject_or(BaseResponder)
+                if responder:
+                    report = ProblemReport(
+                        description={
+                            "en": (
+                                "Presentation request not properly formatted,"
+                                " TypeError raised on Holder agent."
+                            ),
+                            "code": ProblemReportReason.ABANDONED.value,
+                        }
                     )
-                    # Defaults to page_size but would like to include all
-                    # For now, setting to 1000
-                    max_results = 1000
-                    records = await search.fetch(max_results)
-                # Avoiding addition of duplicate records
-                (
-                    vcrecord_list,
-                    vcrecord_ids_set,
-                ) = await self.process_vcrecords_return_list(records, record_ids)
-                record_ids = vcrecord_ids_set
-                credentials_list = credentials_list + vcrecord_list
-        except StorageNotFoundError as err:
-            raise V20PresFormatHandlerError(err)
-        except TypeError as err:
-            LOGGER.error(str(err))
-            responder = self._profile.inject_or(BaseResponder)
-            if responder:
-                report = ProblemReport(
-                    description={
-                        "en": (
-                            "Presentation request not properly formatted,"
-                            " TypeError raised on Holder agent."
-                        ),
-                        "code": ProblemReportReason.ABANDONED.value,
-                    }
-                )
-                if pres_ex_record.thread_id:
-                    report.assign_thread_id(pres_ex_record.thread_id)
-                await responder.send_reply(
-                    report, connection_id=pres_ex_record.connection_id
-                )
-                return
-
+                    if pres_ex_record.thread_id:
+                        report.assign_thread_id(pres_ex_record.thread_id)
+                    await responder.send_reply(
+                        report, connection_id=pres_ex_record.connection_id
+                    )
+                    return
+        elif claim_fmt.di_vc:
+            holder = AnonCredsHolder(self._profile)
```

Comment:
hey @ianco we are discussing the steps, there is some confusion about which methods on AnonCredsHolder to use, we need to call https://github.com/Whats-Cookin/aries-cloudagent-python/blob/55d581e47e410b43f486e86062e61f040fc4a5ca/aries_cloudagent/anoncreds/holder.py#L363 ? 

@sarthakvijayvergiya do you want to share the ways you did alraedy try calling this, and what were the results?

Reply:
This is the call to get credentials that can satisfy a proof request.  I haven't checked but I assume you need to call this after you've converted the DIF proof request to a "legacy" proof request.